### PR TITLE
update bindgen to fix incorrect proj results on apple silicon

### LINKED ISSUE #52: Tests fail on Apple Silicon
```
% cargo test
    Finished test [unoptimized + debuginfo] target(s) in 0.08s
     Running target/debug/deps/proj-6a37e3d3cad5067a

running 14 tests
test proj::test::test_definition ... ok
test proj::test::test_conversion_error ... FAILED
test proj::test::test_conversion ... FAILED
test proj::test::test_error_recovery ... FAILED
test proj::test::test_inverse_projection ... FAILED
test proj::test::test_london_inverse ... FAILED
test proj::test::test_searchpath ... ok
test proj::test::test_projection ... FAILED
test proj::test::test_area_of_use ... ok
test proj::test::test_array_convert ... ok
test proj::test::test_from_crs ... FAILED
test proj::test::test_input_order ... FAILED
test proj::test::test_set_endpoint ... ok
proj_create: unrecognized format / unknown name
test proj::test::test_init_error ... ok

failures:

---- proj::test::test_conversion_error stdout ----
thread 'proj::test::test_conversion_error' panicked at 'called `Result::unwrap_err()` on an `Ok` value: MyPoint { x: 0.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000021451195286, y: 0.00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000057 }', src/proj.rs:1057:14

---- proj::test::test_conversion stdout ----
thread 'proj::test::test_conversion' panicked at 'assertion failed: f > 0.99999', src/proj.rs:885:9

---- proj::test::test_error_recovery stdout ----
thread 'proj::test::test_error_recovery' panicked at 'assertion failed: nad83_m.convert(MyPoint::new(0.0, 0.0)).is_ok()', src/proj.rs:1077:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- proj::test::test_inverse_projection stdout ----
thread 'proj::test::test_inverse_projection' panicked at 'assertion failed: f > 0.99999', src/proj.rs:885:9

---- proj::test::test_london_inverse stdout ----
thread 'proj::test::test_london_inverse' panicked at 'assertion failed: f > 0.99999', src/proj.rs:885:9

---- proj::test::test_projection stdout ----
thread 'proj::test::test_projection' panicked at 'assertion failed: f > 0.99999', src/proj.rs:885:9

---- proj::test::test_from_crs stdout ----
thread 'proj::test::test_from_crs' panicked at 'assertion failed: f > 0.99999', src/proj.rs:885:9

---- proj::test::test_input_order stdout ----
thread 'proj::test::test_input_order' panicked at 'assertion failed: `(left == right)`
  left: `6693625.67217475`,
 right: `0.00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000005`', src/proj.rs:1110:9


failures:
    proj::test::test_conversion
    proj::test::test_conversion_error
    proj::test::test_error_recovery
    proj::test::test_from_crs
    proj::test::test_input_order
    proj::test::test_inverse_projection
    proj::test::test_london_inverse
    proj::test::test_projection

test result: FAILED. 6 passed; 8 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.12s

error: test failed, to rerun pass '--lib'
```

### PR DESCRIPTION
**Pull request by michaelkirk**
FIXES #52

### CODE REVIEW

**Comment by lnicola** [comment]
Should we also regenerate https://github.com/georust/proj/blob/master/proj-sys/src/bindings_docs-rs.rs?

**Comment by michaelkirk** [comment]
Excellent observation, thank you @lnicola. 

done!

File: proj-sys/src/bindings_docs-rs.rs [hunk 1]
```diff
@@ -1,10 +1,10 @@
 /* THESE ARE NOT LIVE BINDINGS */
 /* THEY EXIST FOR USE BY DOCS-RS ONLY */
-/* automatically generated by rust-bindgen */
+/* automatically generated by rust-bindgen 0.58.1 */
 
 pub const PROJ_VERSION_MAJOR: u32 = 7;
-pub const PROJ_VERSION_MINOR: u32 = 1;
-pub const PROJ_VERSION_PATCH: u32 = 0;
+pub const PROJ_VERSION_MINOR: u32 = 2;
```
Comment (michaelkirk) [inline proj-sys/src/bindings_docs-rs.rs:6]: Docs were actually not updated last time we updated libproj.

**Review by michaelkirk — COMMENTED** [review]
> It would be really nice to not require bindgen as it's a heavy dependency. gdal-sys actually has some pre-built bindings for this. But now I wonder how portable those bindings actually are.

I'm with you. #44 

I think we're onboard. Just no-one has done it yet.

> 🔒 Permission denied

I've just now added you! Can you try again?

File: proj-sys/src/bindings_docs-rs.rs [hunk 2]
```diff
@@ -1744,6 +1785,18 @@ pub type proj_network_get_header_value_cbk_type = ::std::option::Option<
...
+#[doc = " Read size_to_read bytes from handle, starting at offset, into"]
+#[doc = " buffer."]
+#[doc = " During this read, the implementation should make sure to store the HTTP"]
+#[doc = " headers from the server response to be able to respond to"]
+#[doc = " proj_network_get_header_value_cbk_type callback."]
+#[doc = ""]
+#[doc = " error_string_max_size should be the maximum size that can be written into"]
+#[doc = " the out_error_string buffer (including terminating nul character)."]
+#[doc = ""]
+#[doc = " @return the number of bytes actually read (0 in case of error)"]
```
[the remark this answers is not in the document]
Comment (lnicola) [reply proj-sys/src/bindings_docs-rs.rs:1799]: See also https://github.com/rust-lang/rust-bindgen/issues/1265

[the remark this answers is not in the document]
Comment (michaelkirk) [reply proj-sys/src/bindings_docs-rs.rs:1799]: lol you're EVERYWHERE.

[the remark this answers is not in the document]
Comment (lnicola) [reply proj-sys/src/bindings_docs-rs.rs:1799]: Yeah, I ran into that in https://github.com/georust/gdal/pull/55.

**Pull request merged**